torch/_cuda_sanitizer/_sanitizer.py

```diff
@@ -0,0 +1,490 @@
+import enum
+import functools
+import io
+import logging
+import sys
+import textwrap
+import traceback
+from dataclasses import dataclass, field
+from typing import Any, Dict, Iterator, List, Optional, Set, Tuple, TypeVar
+
+import torch
+import torch.utils._cuda_trace as cuda_trace
+from torch.utils._cuda_trace import _CUDA_HOOK_TRACE_OFFSET
+from torch.utils._python_dispatch import (
+    TorchDispatchMode,
+    _PYTHON_DISPATCH_TRACE_OFFSET,
+)
+from torch.utils._pytree import tree_map
+
+
+TK = TypeVar("TK")
+TVa = TypeVar("TVa")
+TVb = TypeVar("TVb")
+
+DataPtr = int
+StreamId = int
+EventId = int
+SeqNum = int
+
+logger = logging.getLogger(__name__)
+
+
+class AccessType(enum.Enum):
+    READ = enum.auto()
+    WRITE = enum.auto()
+
+    def __str__(self):
+        return "reading from" if self is AccessType.READ else "writing to"
+
+
+@dataclass
```

Comment:
Comment saying what this class is!  In particular, it would be helpful to know what the semantics of "operator" and "names" are; they are non-obvious from the types.